x/marker/simulation/genesis.go

```diff
@@ -17,15 +19,22 @@ import (
 // Simulation parameter constants
 const (
 	MaxTotalSupply         = "max_total_supply"
+	MaxSupply              = "max_supply"
 	EnableGovernance       = "enable_governance"
 	UnrestrictedDenomRegex = "unresticted_denom_regex"
 )
 
-// GenMaxTotalSupply randomized Maximum amount of supply to allow for markers
+// GenMaxTotalSupply is deprecated.
+// Deprecated: GenMaxTotalSupply is kept for backwards compatibility.
 func GenMaxTotalSupply(r *rand.Rand) uint64 {
 	return r.Uint64()
 }
 
+// GenMaxSupply randomized Maximum amount of supply to allow for markers
+func GenMaxSupply(r *rand.Rand) math.Int {
+	return math.NewIntFromUint64(r.Uint64())
```

Comment:
We should expand this range to verify that the system works as intended with the new `> uint64` maximum range.